### PR TITLE
KAFKA-13284: Use sftp to upload artifacts in release.py

### PR DESCRIPTION
**Pull request by mimaison**
- automatically upload artifacts to Apache Home using sftp
- fix the logic to find JDK 17

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: release.py [hunk 1]
```diff
@@ -207,10 +217,10 @@ def get_jdk(prefs, version):
     jdk_java_home = get_pref(prefs, 'jdk%d' % version, lambda: raw_input("Enter the path for JAVA_HOME for a JDK%d compiler (blank to use default JAVA_HOME): " % version))
     jdk_env = dict(os.environ) if jdk_java_home.strip() else None
     if jdk_env is not None: jdk_env['JAVA_HOME'] = jdk_java_home
-    javaVersion = cmd_output("%s/bin/java -version" % jdk_java_home, env=jdk_env)
-    if version == 8 and "1.8.0" not in javaVersion:
+    java_version = cmd_output("%s/bin/java -version" % jdk_java_home, env=jdk_env)
+    if version == 8 and "1.8.0" not in java_version:
       fail("JDK 8 is required")
-    elif "%d.0" % version not in javaVersion:
+    elif "%d" % version not in java_version:
```
Comment (mimaison) [inline release.py]: OpenJDK 17 version:
```
openjdk version "17" 2021-09-14
OpenJDK Runtime Environment Temurin-17+35 (build 17+35)
OpenJDK 64-Bit Server VM Temurin-17+35 (build 17+35, mixed mode, sharing)
```

File: release.py [hunk 2]
```diff
@@ -621,7 +631,8 @@ def select_gpg_key():
 cmd("Listing artifacts to be uploaded:", "ls -R %s" % artifacts_dir)
 
 cmd("Zipping artifacts", "tar -czf %s.tar.gz %s" % (artifact_name, artifact_name), cwd=work_dir)
-if not user_ok("Have you uploaded artifacts in %s, listed above, to public_html in your Apache home directory (y/n)?: " % artifacts_dir):
+sftp_upload(artifacts_dir)
```
Comment (dajac) [inline release.py:632]: Should we also call `sftp_mkdir` before to ensure that the directory exists?

File: release.py [hunk 3]
```diff
@@ -178,6 +178,16 @@ def sftp_mkdir(dir):
         # This is ok. The command fails if the directory already exists
         pass
 
+def sftp_upload(dir):
+    try:
+       cmd_str  = """
+cd public_html
+put -r %s
+""" % dir
+       cmd("Uploading '%s' under public_html in your Apache home directory" % dir, "sftp -b - %s@home.apache.org" % apache_id, stdin=cmd_str, allow_failure=True, num_retries=3)
```
Comment (dajac) [inline release.py]: I guess that we assume that the relevant key is in the ssh-agent, right?

**Review by dajac — COMMENTED** [review]
Thanks for the PR. I left a few comments.

File: release.py (hunk 3 above)
Comment (mimaison) [reply release.py]: Yes like it used to work with rsync

File: release.py (hunk 2 above)
Comment (dajac) [reply release.py:632]: I meant `public_html` directory.

Comment (mimaison) [reply release.py:632]: Yes that's what I understood. I've made the changes, testing it now ...

Comment (mimaison) [reply release.py:632]: Thanks for the suggestion, I pushed an update

**Review by dajac — APPROVED** [review]
LGTM, thanks!

**Pull request merged**